pkg/destroy/aws/aws.go

```diff
@@ -1597,18 +1601,40 @@ func deleteS3(session *session.Session, arn arn.ARN, logger logrus.FieldLogger)
 		Bucket: aws.String(arn.Resource),
 	})
 	err := s3manager.NewBatchDeleteWithClient(client).Delete(aws.BackgroundContext(), iter)
-	if err != nil {
+	if err != nil && !isBucketNotFound(err) {
 		return err
 	}
 	logger.Debug("Emptied")
 
 	_, err = client.DeleteBucket(&s3.DeleteBucketInput{
 		Bucket: aws.String(arn.Resource),
 	})
-	if err != nil {
+	if err != nil && !isBucketNotFound(err) {
 		return err
 	}
 
 	logger.Info("Deleted")
 	return nil
 }
+
+func isBucketNotFound(err interface{}) bool {
+	switch s3Err := err.(type) {
+	case awserr.Error:
+		if s3Err.Code() == bucketNotFound {
+			return true
+		}
+		origErr := s3Err.OrigErr()
+		if origErr != nil {
+			return isBucketNotFound(origErr)
+		}
+	case s3manager.Error:
+		if s3Err.OrigErr != nil {
+			return isBucketNotFound(s3Err.OrigErr)
+		}
+	case s3manager.Errors:
```

Comment:
If it's a list of errors, why not check to make sure one of them is not found..?

Reply:
the only case where we want to skip an error from this is when a bucket is not found. That is returned as a slice of errors with one member. If you have multiple errors and only one of them is NotFound, you don't want to declare the delete successful.

Reply:
+1